pennylane/grouping/utils.py

```diff
@@ -333,6 +337,230 @@ def binary_to_pauli(binary_vector, wire_map=None):  # pylint: disable=too-many-b
+
+    pauli_word = None
+
+    for wire_name, wire_idx in wire_map.items():
+        pauli_char = pauli_string[wire_idx]
+
+        # Don't care about the identity
+        if pauli_char == "I":
+            continue
+
+        if pauli_word is not None:
+            pauli_word = pauli_word @ character_map[pauli_char](wire_name)
+        else:
+            pauli_word = character_map[pauli_char](wire_name)
+
+    return pauli_word
+
+
+def pauli_word_to_matrix(pauli_word, wire_map=None):
+    """Convert a Pauli operation from a tensor to its matrix representation.
+
+    The matrix representation of a Pauli word has dimension :math:`2^n \times 2^n`,
+    where :math:`n` is the number of qubits provided in ``wire_map``. For wires
+    that the Pauli word does not act on, identities must be inserted into the tensor
+    product at the correct positions.
+
+    Args:
+        pauli_word (Observable): an observable, either a :class:`~.Tensor` instance or
+            single-qubit observable representing a Pauli group element.
+        wire_map (dict[Union[str, int], int]): dictionary containing all wire labels used in
+            the Pauli word as keys, and unique integer labels as their values
+
+    Returns:
+        (np.ndarray): The matrix representation of the multi-qubit Pauli over the
+            specified wire map.
+
+    Raises:
+        TypeError: if the input observable is not a proper Pauli word.
+    """
+    if not is_pauli_word(pauli_word):
+        raise TypeError(
+            "Expected a Pauli word Observable instance, instead got {}.".format(pauli_word)
+        )
+
+    # If there is no wire map, we must infer from the structure of Paulis
+    if wire_map is None:
+        wire_map = {pauli_word.wires.labels[i]: i for i in range(len(pauli_word.wires))}
+
+    n_qubits = len(wire_map)
+
+    # If there is only a single qubit, we can return the matrix directly
+    if n_qubits == 1:
+        return pauli_word.matrix
+
+    # If there is more than one qubit, we must go through the wire map wire
+    # by wire and pick out the relevant matrices
+    pauli_mats = [np.eye(2) for x in range(n_qubits)]
+
+    # There may be more than one qubit in the Pauli but still only
+    # one of them with anything acting on it, so take that into account
+    pauli_names = [pauli_word.name] if isinstance(pauli_word.name, str) else pauli_word.name
+
+    for wire_label, wire_idx in wire_map.items():
+        if wire_label in pauli_word.wires.labels:
+            op_idx = pauli_word.wires.labels.index(wire_label)
+
+            if pauli_names[op_idx] == "PauliX":
+                pauli_mats[wire_idx] = PauliX.matrix
+            elif pauli_names[op_idx] == "PauliY":
+                pauli_mats[wire_idx] = PauliY.matrix
+            else:
+                pauli_mats[wire_idx] = PauliZ.matrix
+
+    return reduce(np.kron, pauli_mats)
+
+
+def are_commuting(pauli_word_1, pauli_word_2, wire_map=None):
+    r"""Checks if two Pauli words commute.
```

Comment:
This is more general than [`is_qwc`](https://pennylane--1159.org.readthedocs.build/en/1159/code/api/pennylane.grouping.is_qwc.html) right?

Tangent: should this be called `is_commuting` 🤔

Reply:
I was debating that but that grammar is not correct! Instead I think it'd be better to rename `is_qwc` to `are_qwc`.

Reply:
is it one of those cases where turning it into an acronym affects whether it is plural? 🤔

Reply:
It's just that it's comparing two things (Paulis) in both cases, so "are they commuting", and "are they qubit-wise commuting".

Reply:
right, but comparing two things in most programming languages is `isequal`, rather than `areequal` 😆 So the first one reads as more correct in my head. But yeah this is a very minor point lol